go/4_auth-aws.go

```diff
@@ -1,17 +1,17 @@
 package main
 
 import (
+	"context"
 	"fmt"
-	"os"
 
-	"github.com/hashicorp/go-hclog"
-	"github.com/hashicorp/go-secure-stdlib/awsutil"
 	vault "github.com/hashicorp/vault/api"
+	auth "github.com/hashicorp/vault/api/auth/aws"
```

Comment:
Just curious, if you think this should be something like `awsAuth` vs. just `auth` b/c it only encompasses the aws part of this

Reply:
I thought about it but I didn't have a good name for the value returned by `NewAWSAuth` so I decided to name that variable awsAuth and this one just auth. I also feel like it kind of shows that each of these "/api/auth/something" packages all work the same way as each other, just with different options in their constructors.

Reply:
That's a good point! I honestly couldn't come up w/ a better name for it either.